File: src/main/java/com/github/morningzeng/toolset/annotations/ScratchConfig.java
package com.github.morningzeng.toolset.annotations;

import com.github.morningzeng.toolset.enums.OutputType;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author Morning Zeng
 * @since 2024-06-28
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface ScratchConfig {

    String value();

    String directory() default "";

    OutputType outputType() default OutputType.YAML;

}

File: src/main/kotlin/com/github/morningzeng/toolbox/annotations/ScratchConfig.kt
package com.github.morningzeng.toolbox.annotations

import com.github.morningzeng.toolbox.enums.JacksonType

/**
 * @author Morning Zeng
 * @since 2025-05-16
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class ScratchConfig(
    val value: String,
    val directory: String = "",
    val outputType: JacksonType = JacksonType.YAML
)